lib/matplotlib/figure.py

```diff
@@ -682,11 +682,10 @@ def set_size_inches(self, *args, **kwargs):
         matplotlib.Figure.get_size_inches
         """
 
-        forward = kwargs.get('forward', False)
-        if len(args) == 1:
-            w, h = args[0]
-        else:
-            w, h = args
+        # the width and height have been passed in as a tuple to the first
```

Comment:
Yeah, I just made it so it would be backwards compatible for how it works currently, but I agree it's kind of strange to support both the tuple and the arguments. I don't know if anything actually uses the tuple notation for this.

I personally use `set_figwidth` and `set_figheight` a lot, which is how this came up, but that's also partially because I didn't know about `set_size_inches`. I like the suggestion of having both width and height be keyword arguments in `set_size_inches` and allow setting either or both, and then deprecating `set_figwidth` and `set_figheight`. Is there a reason, though, for it being `set_size_inches` rather than just `set_size`?